tICA/scripts/ComputeGroupTICA.m

```diff
@@ -0,0 +1,247 @@
+
+    %tica runwise normalization
+    TCSFullRunVars = single(zeros(sICAdim, RunsXNumTimePoints * numsubj));
+    for i = 1:numsubj
+        if exist(TCSList{i}, 'file')
+            subjBaseInd = (i - 1) * RunsXNumTimePoints + 1;
+            thisStart = subjBaseInd;
+            TCSRunVarSub = [];
+            SubjFolder = [StudyFolder '/' SubjectList{i} '/'];
+            for j = 1:length(fMRIList)
+                dtseriesName = [SubjFolder fMRIList{j}];
+                if exist(dtseriesName, 'file')
+                    [~, runLengthStr] = system(['wb_command -file-information -only-number-of-maps ' dtseriesName]);
+                    runLength = str2double(runLengthStr);
+                    nextStart = thisStart + runLength;
+                    TCSRunVarSub = [TCSRunVarSub repmat(std(TCSFullConcat.cdata(:, thisStart:(nextStart - 1)), [], 2), 1, runLength)];
+
+                    thisStart = nextStart;
+                end
+            end
+            if size(TCSRunVarSub, 2) == RunsXNumTimePoints
+                numfullsubj = numfullsubj + 1;
+            end
+            TCSFullRunVars(:, subjBaseInd - 1 + (1:size(TCSRunVarSub, 2))) = TCSRunVarSub;
+        end
+    end
+    %end tica runwise normalization
+
+    nlfunc = 'tanh';
+    iterations = 100;
+    tICAdim = sICAdim;
+
+    sICAtcsvars = std(TCSFullConcat.cdata, [], 2);
+    TCSFullConcat.cdata = (TCSFullConcat.cdata ./ TCSFullRunVars) .* repmat(sICAtcsvars, 1, size(TCSFullRunVars, 2)); %Making all runs contribute equally improves tICA decompositions
+    TCSFullConcat.cdata(~isfinite(TCSFullConcat.cdata)) = 0;
+
+    %This loop produces more reproducible and better tICA decompositions
+    if strcmp(tICAmode,'ESTIMATE')
+        ITERATIONS=[1:6];
+    elseif strcmp(tICAmode,'INITIALIZE')
+        ITERATIONS=[2:6];
+        A = tICAMM;
+        if size(tICAMM,1) ~= size(TCSFullConcat.cdata,1)
+            error('Initialization mixing matrix does not match dimensions of the sICA components');
+        end
+    elseif strcmp(tICAmode,'USE')
+        ITERATIONS=[0];
+        A = tICAMM;
+        if size(tICAMM,1) ~= size(TCSFullConcat.cdata,1)
+            error('Mixing matrix to be used does not match dimensions of the sICA components');
+        end
+    else
+        error('tICAmode not recognized');
+    end
+    
+    for i = ITERATIONS
+        if  i == 0
+            IT = ['F'];
+            W = pinv(A);
+            normicasig = W * TCSFullConcat.cdata;
+        elseif i == 1
+            IT = [num2str(i)];
+            [iq, A, W, normicasig, sR] = icasso('both', TCSFullConcat.cdata, iterations, 'approach', 'symm', 'g', nlfunc, 'lastEig', sICAdim, 'numOfIC', tICAdim, 'maxNumIterations', 1000); %x1
+        elseif i > 5
+            IT = ['F'];
+            [normicasig, A, W] = fastica(TCSFullConcat.cdata, 'initGuess', A, 'approach', 'symm', 'g', nlfunc, 'lastEig', sICAdim, 'numOfIC', tICAdim, 'displayMode', 'off', 'maxNumIterations', 1000); %x1
+        else
```

Comment:
This looks like it will never get executed, `ITERATIONS` will never contain a negative number due to the above `elseif` tree setting it (or erroring) in all branches.